include/deal.II/numerics/vector_tools_project.templates.h

```diff
@@ -773,19 +775,25 @@ namespace VectorTools
      * Check if we can use the MatrixFree implementation or need
      * to use the matrix based one.
      */
-    template <typename VectorType, int dim>
-    void
+    template <typename VectorType, int dim, int spacedim>
+    typename std::enable_if<
+      !numbers::NumberTraits<typename VectorType::value_type>::is_complex &&
+        dim == spacedim,
+      void>::type
     project(
-      const Mapping<dim> &                                      mapping,
-      const DoFHandler<dim> &                                   dof,
-      const AffineConstraints<typename VectorType::value_type> &constraints,
-      const Quadrature<dim> &                                   quadrature,
-      const Function<dim, typename VectorType::value_type> &    function,
-      VectorType &                                              vec_result,
+      const Mapping<dim, spacedim> &                             mapping,
+      const DoFHandler<dim, spacedim> &                          dof,
+      const AffineConstraints<typename VectorType::value_type> & constraints,
+      const Quadrature<dim> &                                    quadrature,
+      const Function<spacedim, typename VectorType::value_type> &function,
+      VectorType &                                               vec_result,
       const bool                 enforce_zero_boundary,
       const Quadrature<dim - 1> &q_boundary,
       const bool                 project_to_boundary_first)
     {
+      static_assert(
+        !numbers::NumberTraits<typename VectorType::value_type>::is_complex,
+        "Complex numbers are not supported in this code path");
```

Comment:
But that can't happen due to the `enable_if`, can it?

Reply:
> But that can't happen due to the enable_if, can it?

You are right, let me remove it again, I used it to convince myself about it.